docs/src/user_docs/assembly/code_organization.md

```diff
@@ -48,20 +48,25 @@ Finally, a procedure cannot contain *solely* any number of [advice injectors](./
 #### Dynamic procedure invocation
 It is also possible to invoke procedures dynamically - i.e., without specifying target procedure labels at compile time. Unlike static procedure invocation, recursion is technically possible using dynamic invocation, but dynamic invocation is more expensive, and has less available operand stack capacity for procedure arguments, as 4 elements are required for the MAST root of the callee. There are two instructions, `dynexec` and `dyncall`, which can be used to execute dynamically-specified code targets. Both instructions expect the [MAST root](../../design/programs.md) of the target to be provided via the stack. The difference between `dynexec` and `dyncall` corresponds to the difference between `exec` and `call`, see the documentation on [procedure invocation semantics](./execution_contexts.md#procedure-invocation-semantics) for more detail.
```

Comment:
> Unlike static procedure invocation, recursion is technically possible using dynamic invocation, but dynamic invocation is more expensive,

Why is recursion only possible using dynamic invocation? Also AFAIU dynamic invocation is not more expensive than static invocation (modulo maybe a memory read)?

Reply:
> Why is recursion only possible using dynamic invocation?

Because a procedure could dynamically invoke itself via memory pointer. Without dynamic invocation recursion is not possible.

> Also AFAIU dynamic invocation is not more expensive than static invocation (modulo maybe a memory read)?

Yeah, not any more. One can argue that there is an extra memory read - but that's negligible.

Reply:
Brought the part about recursion back but reworded slightly